CSharpSourceCode/AttributeDataSystem/StaticAttributeMissionLogic.cs

```diff
@@ -0,0 +1,125 @@
+        
+        private List<PartyAttribute> defenderAttributes;
+
+        private List<Agent> _agents;
+        
+        public override void AfterStart()
+        {
+            base.AfterStart();
+            teams= Mission.Current.Teams;
+            activeAttributes = new List<PartyAttribute>();
+            defenderAttributes = new List<PartyAttribute>();
+            _agents = new List<Agent>();
+            
+            if (Campaign.Current != null)
+            {
+                _partyAttributeManager= PartyAttributeManager.Instance;
+                activeAttributes = _partyAttributeManager.GetActiveInvolvedParties();
+            }
+            
+        }
+        
+         public override void OnAgentBuild(Agent agent, Banner banner)
+        {
+            base.OnAgentBuild(agent, banner);
+
+            if (agent.IsMount)
+                return;
+            
+            foreach (var partyAttribute in activeAttributes)
+            {
+                if (agent.Origin.BattleCombatant== partyAttribute.PartyBase)
+                {
+                    if (partyAttribute.PartyType == PartyType.RogueParty)
+                    {
+                        foreach (var attribute in partyAttribute.RegularTroopAttributes)
+                        {
+                            AddStaticAttributeComponent(agent, attribute, partyAttribute);
+                            break;
+                        }
+                        break;
+                    }
+
+                    if (partyAttribute.PartyType == PartyType.Regular)
+                    {
+                        foreach (var attribute in partyAttribute.RegularTroopAttributes)
+                        {
+                            if (agent.Origin.Troop.ToString() == attribute.id)
+                            {
+                                AddStaticAttributeComponent(agent, attribute, partyAttribute);
+                                break;
+                            }
+                        }
+                        break;
+                    }
+
+                    if (partyAttribute.PartyType == PartyType.LordParty)
+                    {
+                        if (!agent.IsHero)
+                        {
+                            if (agent.Character.IsSoldier && !partyAttribute.RegularTroopAttributes.IsEmpty())
+                            {
+                                foreach (var attribute in partyAttribute.RegularTroopAttributes)
+                                {
+                                    if (agent.Character.ToString() == attribute.id)
+                                    {
+                                        AddStaticAttributeComponent(agent, attribute, partyAttribute);
+                                        break;
+                                    }
+
+                                }
+                                break;
+                            }
+                        }
+                        else
```

Comment:
This `else` contains two `if` blocks with no shared logic before or after them. It'd be cleaner to just have 

```
else if (agent.Character.Name == partyAttribute.Leader.Name)
{
    var leaderAttribute = partyAttribute.LeaderAttribute;
    AddStaticAttributeComponent(agent, leaderAttribute, partyAttribute);
    break;
}
else if (!partyAttribute.CompanionAttributes.IsEmpty())
{
    foreach (var companionAttribute in partyAttribute.CompanionAttributes)
    {
        if (agent.Character.Name.ToString() == companionAttribute.id)
        {
            AddStaticAttributeComponent(agent, companionAttribute, partyAttribute);
            break;
        }
    }
}
```